JavaScript/chapter01/1.2 - Check Perm/rroque98_sol.js

```diff
@@ -15,15 +15,14 @@ const isPermutation = (str1, str2) =>{
   return true;
   //******** helper function ******
   function determineCharCount(string) {
-    const obj = {};
-    for (let char of string) {
-      if (obj[char] !== undefined) {
-        obj[char]++;
+    return string.split('').reduce((acc, char) => {
+      if (acc[char] !== undefined) {
```

Comment:
it might be a bit better to use a [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) like you do in https://github.com/techqueria/data-and-algorithms/pull/26/files#diff-440386f91fbaa9553b4d4652c2d527acR8

Since [`Map.set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/set) supports chaining, you can just do

```javascript
return string
    .split('')
    .reduce((acc, char) => acc.set(char, (acc.get(char) || 0) + 1),
            new Map());
```

Reply:
Thank you for the feedback. I just submitted the changes to use Map instead